lib/cartopy/img_transform.py

```diff
@@ -358,7 +358,12 @@ def regrid(array, source_x_coords, source_y_coords, source_cs, target_proj,
         if np.ma.isMaskedArray(new_array):
             new_array[non_self_inverse_points] = np.ma.masked
         else:
-            new_array = np.ma.array(new_array, mask=non_self_inverse_points)
+            new_array = np.ma.array(new_array, mask=False)
+            if new_array.ndim == 3:
+                for i in range(new_array.shape[2]):
+                    new_array[non_self_inverse_points, i] = np.ma.masked
```

Comment:
Is `non_self_inverse_points` 2D? I don't understand why the loop range is using the third dimension, but this indexing is using the second. And why can't you just use `:` instead of the loop?